src/platform/auth.ts

```diff
@@ -179,76 +183,77 @@ export function setUserTokenDataCallback(fn: GetUserTokenDataFn) {
+	userId,
+	existingToken,
+): Promise<User> => {
+	const [userData, permissionData] = await Promise.all([
+		api.resin.get({
+			resource: 'user',
+			id: userId,
+			passthrough: { req: root },
+			options: {
+				$select: tokenFields,
+			},
+		}) as Promise<AnyObject>,
+		sbvrUtils.getUserPermissions(userId),
+	]);
+	// (userData: AnyObject, permissionData) => {
+	if (!userData || !permissionData) {
+		throw new Error('No data found?!');
+	}
+	const newTokenData: Partial<User> = _.pick(userData, tokenFields);
 
-	const permissionData = sbvrUtils.getUserPermissions(userId);
-
-	return Promise.join(
-		userData,
-		permissionData,
-		(userData: AnyObject, permissionData) => {
-			if (!userData || !permissionData) {
-				throw new Error('No data found?!');
-			}
-			const newTokenData: Partial<User> = _.pick(userData, tokenFields);
-
-			const tokenData = {
-				...existingToken,
-				...newTokenData,
-				permissions: permissionData,
-			} as User;
-
-			if (!_.isFinite(tokenData.authTime)) {
-				tokenData.authTime = Date.now();
-			}
-
-			// skip nullish attributes
-			return _.omitBy(tokenData, _.isNil) as User;
-		},
-	);
+	const tokenData = {
+		...existingToken,
+		...newTokenData,
+		permissions: permissionData,
+	} as User;
+
+	if (!_.isFinite(tokenData.authTime)) {
+		tokenData.authTime = Date.now();
+	}
+
+	// skip nullish attributes
+	return _.omitBy(tokenData, _.isNil) as User;
 };
 
-export const createSessionToken = (
+export const createSessionToken = async (
 	userId: number,
 	{ existingToken, jwtOptions }: ExtraParams = {},
 ): Promise<string> => {
-	return Promise.resolve(
-		$getUserTokenDataCallback(userId, existingToken),
-	).then(tokenData => createJwt(tokenData, jwtOptions));
+	const tokenData = await $getUserTokenDataCallback(userId, existingToken);
+	return createJwt(tokenData, jwtOptions);
 };
 
-const sendXHRToken = (res: Response, statusCode = 200) => (
-	token: string,
-): void => {
+const sendXHRToken = (res: Response, token: string, statusCode = 200): void => {
 	res.header('content-type', 'text/plain');
 	res.status(statusCode).send(token);
 };
 
-export const loginUserXHR = (
+export const loginUserXHR = async (
 	res: Response,
 	userId: number,
 	statusCode?: number,
 	extraParams?: ExtraParams,
-): Promise<void> =>
-	createSessionToken(userId, extraParams).then(sendXHRToken(res, statusCode));
-
-export const updateUserXHR = (res: Response, req: Request): Promise<void> =>
-	getUser(req, false)
-		.then(() => {
-			if (req.creds == null || !('id' in req.creds) || req.creds.id == null) {
-				throw new InternalRequestError('No user present');
-			}
-			return createSessionToken(req.creds.id, { existingToken: req.creds });
-		})
-		.then(sendXHRToken(res));
+): Promise<void> => {
+	const token = await createSessionToken(userId, extraParams);
+	sendXHRToken(res, token, statusCode);
+};
+
+export const updateUserXHR = async (
+	res: Response,
+	req: Request,
+): Promise<void> => {
+	await getUser(req, false);
```

Comment:
Let's add a comment here that we care about `getUser`'s side-effects. Otherwise, the fact we're passing `false` for `required` suggests we don't care if this fails which makes one wonder why this call is useful here.